opencog/spacetime/pointmemory/PointMemorySCM.cc

```diff
@@ -325,10 +325,15 @@ Handle PointMemorySCM::map_ato(Handle map_name, Handle ato, Handle loc)
 	have_map(map_name);
 	// loc should be a ListLink of three NumberNodes.
 	const HandleSeq& hs = loc->getOutgoingSet();
-	double x = NumberNodeCast(hs[0])->get_value();
-	double y = NumberNodeCast(hs[1])->get_value();
-	double z = NumberNodeCast(hs[2])->get_value();
-	tsa[map_name]->insert_atom(point3d(x, y, z), ato);
+	assert(hs.size() == 0 or hs.size() == 3);
+    if(hs.size() == 3) {
```

Comment:
If you really feel that you do need to check the argument, then throw a `RuntimeError`